scipy/linalg/flapack_pos_semidef.pyf.src

```diff
@@ -0,0 +1,48 @@
+! Signatures for f2py-wrappers of FORTRAN LAPACK Symmetric Positive Semidefinite Matrix functions.
+!
+
+subroutine <prefix2>pstrf(n,a,lda,piv,rank_bn,tol,work,info,lower,clean)
+    ! c,x,info = pstrf(a,b,lower=0,overwrite_a=0,overwrite_b=0)
+    ! Solve A * X = B.
+    ! A is symmetric positive semidefinite
+    ! P**T * A * P = U**T * U ,  if UPLO = 'U',
+    ! P**T * A * P = L  * L**T,  if UPLO = 'L',
+    ! where U is an upper triangular matrix and L is lower triangular, and
+    ! P is stored as vector PIV.
+    callstatement (*f2py_func)((lower?"L":"U"),&n,a,&lda,piv,&rank_bn,&tol,work,&info); if(clean){int i,j;if(lower){for(i=0;i\<n;++i) for(j=i+1;j\<n;++j) *(a+j*n+i)=0.0;} else {for(i=0;i\<n;++i) for(j=i+1;j<n;++j) *(a+i*n+j)=0.0;}}
```

Comment:
This is much easier on the python side to cleanup and creates a maintenance burden here.

Reply:
IMHO, cleaning code is best kept at the lowest level.